ios/Capacitor/Capacitor/CapacitorBridge.swift

```diff
@@ -537,6 +537,60 @@ internal class CapacitorBridge: NSObject, CAPBridgeProtocol {
         }
     }
 
+    // MARK: - CAPBridgeProtocol: Paths, Files, Assets
+
+    /**
+     * Translate a URL from the web view into a file URL for native iOS.
+     *
+     * The web view may be handling several different types of URLs:
+     *   - res:// (shortcut scheme to web assets)
+     *   - file:// (fully qualified URL to file on the local device)
+     *   - base64:// (to be implemented)
+     *   - [web view scheme]:// (already converted once to load in the web view, to be implemented)
+     */
+    public func localURL(fromWebURL webURL: URL?) -> URL? {
+        guard let inputURL = webURL else {
+            return nil
+        }
+
+        let manager = FileManager.default
+        let url: URL
+
+        switch inputURL.scheme {
+        case "res":
+            let bundle = Bundle.main
+            let resourcePath = bundle.resourcePath!
+            url = URL(fileURLWithPath: resourcePath + "/public" + inputURL.path)
+        case "file":
+            url = inputURL
+        default:
+            return nil
+        }
+
+        guard manager.fileExists(atPath: url.absoluteString.replacingOccurrences(of: "file://", with: "")) else {
+            return nil
+        }
+
+        return url
+    }
+
+    /**
+     * Translate a file URL for native iOS into a URL to load in the web view.
+     */
+    public func portablePath(fromLocalURL localURL: URL?) -> String? {
```

Comment:
@ikeith What do you think about this also returning a proper `URL`, and then documenting that the plugin can use `url.absoluteString`?

Reply:
I like that better.